simulator/infra/config/t/unit_test.cpp

```diff
@@ -0,0 +1,242 @@
+    ASSERT_NO_THROW(config :: handleArgs( argc, const_cast<char**>(argv)));
+
+
+    ASSERT_EQ( config :: num_steps, num_steps);
+    ASSERT_FALSE( binary_filename.compare( config :: binary_filename));
+    ASSERT_EQ( config :: disassembly_on, false);
+    ASSERT_EQ( config :: functional_only, true);
+}
+
+
+
+// 
+// To check whether returned values 
+// are equal to passed arguments
+//
+TEST( config_parse,  Pass_Valid_Args_2)
+{
+    const uint64 num_steps = 356;
+    const std :: string binary_filename{ "run_test.elf"};
+
+    const char* argv[] = 
+    {
+        "mipt-mips",
+        "-b", "run_test.elf",
+        "-n", "356",
+        "-d"
+    };
+    const int argc = static_cast<int>( sizeof( argv) / sizeof( char*));
+    
+    // should not throw any exceptions
+    ASSERT_NO_THROW( config :: handleArgs( argc, const_cast<char**>(argv)));
+
+
+    ASSERT_EQ( config :: num_steps, num_steps);
+    ASSERT_FALSE( binary_filename.compare( config :: binary_filename));
+    ASSERT_EQ( config :: disassembly_on, true);
+    ASSERT_EQ( config :: functional_only, false);
+}
+
+
+
+// 
+// Pass no arguments
+//
+TEST( config_parse,  Pass_No_Args)
+{
+    const char* argv[] = 
+    {
+        "mipt-mips"
+    };
+    const int argc = static_cast<int>( sizeof( argv) / sizeof( char*));
+    
+    // should exit with EXIT_FAILURE
+    ASSERT_EXIT( config :: handleArgs( argc, const_cast<char**>(argv)), 
+                 :: testing :: ExitedWithCode( EXIT_FAILURE), "");
+}
+
+
+
+// 
+// Pass arguments without a binary option
+//
+TEST( config_parse,  Pass_Args_Without_Binary_Option)
+{
+    const char* argv[] = 
+    {
+        "mipt-mips",
+        "--numsteps", "356",
+    };
+    const int argc = static_cast<int>( sizeof( argv) / sizeof( char*));
+    
+    // should exit with EXIT_FAILURE
+    ASSERT_EXIT( config :: handleArgs( argc, const_cast<char**>(argv)), 
+                 :: testing :: ExitedWithCode( EXIT_FAILURE), "");
+}
+
+
+
+// 
+// Pass arguments without a numsteps option
+//
+TEST( config_parse,  Pass_Args_Without_Numsteps_Option)
+{
+    const char* argv[] = 
+    {
+        "mipt-mips",
+        "--binary", "test.elf", 
+    };
+    const int argc = static_cast<int>( sizeof( argv) / sizeof( char*));
+    
+    // should exit with EXIT_FAILURE
+    ASSERT_EXIT( config :: handleArgs( argc, const_cast<char**>(argv)), 
+                 :: testing :: ExitedWithCode( EXIT_FAILURE), "");
+}
+
+
+
+// 
+// Pass arguments with unrecognised option
+//
+TEST( config_parse,  Pass_Args_With_Unrecognised_Option)
+{
+    const char* argv[] = 
+    {
+        "mipt-mips",
+        "--binary", "test.elf",
+        "-n", "356",
+        "-koption"
+    };
+    const int argc = static_cast<int>( sizeof( argv) / sizeof( char*));
+    
+    // should exit with EXIT_FAILURE
+    ASSERT_EXIT( config :: handleArgs( argc, const_cast<char**>(argv)), 
+                 :: testing :: ExitedWithCode( EXIT_FAILURE), "");
+}
+
+
+
```

Comment:
Could you please add a test case when two values of the same option are provided?

Reply:
Sorry, I see it below.